### DISCUSSION TITLE
Subscriptions not firing

### DISCUSSION THREAD

**Discussion by vsoch** [riverqueue/river#490] [Q&A]
I'm noticing that my subscriptions aren't firing beyond the first job returned, and I'm trying to figure out why. I read that they only fire for the specific client they are submit with, and I don't think that's my problem because I only have one. Are there specific things I can look into? The only issue I noticed is that when I left it running for a few hours and came back, the context seemed to have timed out, and I was getting errors from [here](https://github.com/riverqueue/river/blob/master/internal/leadership/elector.go) about not being able to elect a new leader.

But the question is irrespective to that issue - I think it should be the case that I get a notification for each worker. I can see them running and complete, but only the first notifies. I'll keep trying things and thinking, and thank you in advance for the help!

**Comment by vsoch** [comment]
Ah - I think I need a way to keep firing the waitForJob function. In the example https://github.com/riverqueue/river/blob/master/example_subscription_test.go we just need to wait for one job each, and I think that's what I might have missed (or at least this is my current hypothesis)!

**Comment by bgentry** [comment]
Subscriptions only emit events on the client where the event occurs. Are you expecting them to emit events from across all clients?

**Reply by vsoch** [reply]
I'm using the same client to submit (which does in batch)

```go
	count, err := q.riverClient.InsertMany(q.Context, batch)
	if err != nil {
		return err
	}

```
using the same client that I create the workers with. I think it's related to firing the function again, and how and when that channel / function is used but I'm not sure. The examples here just show subscribing for one job, and I'd like to have something that continues running.

**Reply by vsoch** [reply]
oh wait I think I did something really dumb - let me test _that_ hypothesis (much more likely I think)!

**Reply by vsoch** [reply]
yep, fixed! I am so sorry for the noise @bgentry - it was indeed just an issue that it needed to be called multiple times. I had moved the function to a different location, and when I did I dropped the `for {}` that (obviously) is required for it to run again. I assumed the function was the same (it was working before) and was debugging other aspects of the setup. 

bloop bloop bloop. But it's working, I'm so happy! :D Here is a little screenshot of the first _set_ of Kubernetes jobs that are being run in Kubernetes with our queue! 

![image](https://github.com/user-attachments/assets/a21a2596-cb1b-405c-97a9-25a4be4d4ef8)

This library is seriously the coolest. :sunglasses:
